app/models/loan.rb

```diff
@@ -1,20 +1,139 @@
 class Loan < ActiveRecord::Base
-  # include TranslationModule, MediaModule
   include ::Translatable
 
+  # create_table :loans do |t|
+  #   ## base Project fields
+  #   t.references :division, index: true, foreign_key: true
+  #   t.references :organization, index:true, foreign_key: true
+  #   t.string :name
+  #   # translatable: summary
+  #   # translatable: details
+  #   t.references :primary_agent, references: :people
+  #   t.references :secondary_agent, references: :people
+  #   t.integer :status_option_id, index: true
+  #   t.integer :project_type_option_id
+  #   ## distinct Loan fields
+  #   t.integer :loan_type_option_id, index: true
+  #   t.integer :public_level_option_id, index: true
+  #   t.decimal :amount   #note, the default precision is (30,6).  (16,6) is probably sufficient.  worth changing?
+  #   t.references :currency, index: true, foreign_key: true
+  #   t.decimal :rate
+  #   t.integer :length_months
+  #   t.references :representative, references: :people
+  #   t.date :signing_date
+  #   t.date :first_interest_payment_date
+  #   t.date :first_payment_date
+  #   t.date :target_end_date
+  #   t.decimal :projected_return
+  #   ## handle to historical demographic data
+  #   t.references :organization_snapshot
+  #   t.timestamps null: false
+
+
   belongs_to :division
   belongs_to :organization
   belongs_to :primary_agent, class_name: 'Person'
   belongs_to :secondary_agent, class_name: 'Person'
   belongs_to :currency
   belongs_to :representative, class_name: 'Person'
-  # has_many :repayments, :foreign_key => 'LoanID'
+  belongs_to :organization_snapshot
+
+  ##JE todo: for next integration pass
+  # has_many :project_steps, as: :project
 
 
   # define accessor like convenience methods for the fields stored in the Translations table
   attr_translatable :summary, :details
 
 
+  validates :division_id, presence: true
+
+
+  # todo: proper handling needs to be defined, probably a pre-populated and editable display name
+  def name
+    "Project with #{organization.try(:name)}"
+  end
+
+  def status
+    STATUS_OPTIONS.label_for(status_option_id)
+  end
+
+  def loan_type
+    LOAN_TYPE_OPTIONS.label_for(loan_type_option_id)
+  end
+
+  # the special name of a default step to use/create when migrating a log without a step
+  DEFAULT_STEP_NAME = '[default]'
+
+  # creates / reuses a default step when migrating ProjectLogs without a proper owning step
+  # beware, not at all optimized, but sufficient for migration
+  def default_step
+    step = project_steps.select{|s| s.summary == DEFAULT_STEP_NAME}.first
+    unless step
+      logger.info {"default step not found for loan[#{id}] - creating"}
+      step = project_steps.create
+      step.update({summary: DEFAULT_STEP_NAME})
+    end
+    step
+  end
```

Comment:
I thought it might have some generic use as the requirements evolve, since it was a use case that the old system supported..  But if not, then yep, it doesn't really need to live here.